Applications/Utils/ModelPreparation/createNeumannBc.cpp

```diff
@@ -0,0 +1,192 @@
+
+#include "MeshGeoToolsLib/MeshNodeSearcher.h"
+#include "MeshLib/IO/readMeshFromFile.h"
+#include "MeshLib/IO/writeMeshToFile.h"
+#include "MeshLib/convertMeshToGeo.h"
+#include "MeshLib/Mesh.h"
+#include "MeshLib/MeshSubsets.h"
+#include "MeshLib/MeshSurfaceExtraction.h"
+
+#include "NumLib/DOF/LocalToGlobalIndexMap.h"
+
+#include "ProcessLib/ProcessVariable.h"
+int main(int argc, char* argv[])
+{
+    ApplicationsLib::LogogSetup logo_setup;
+
+    TCLAP::CmdLine cmd(
+        "Integrates the given element property and outputs an OGS-5 direct "
+        "Neumann boundary condition",
+        ' ',
+        "0.1");
+
+    TCLAP::ValueArg<std::string> in_mesh("i",
+                                         "in-mesh",
+                                         "the mesh that has an element "
+                                         "property for the Neumann "
+                                         "boundary condition",
+                                         true,
+                                         "",
+                                         "filename for mesh input");
+    cmd.add(in_mesh);
+    TCLAP::ValueArg<std::string> property_arg(
+        "p",
+        "property-name",
+        "name of property for the Neumann boundary condition",
+        true,
+        "",
+        "string (property name)");
+    cmd.add(property_arg);
+
+    TCLAP::ValueArg<std::string> result_file(
+        "o",
+        "result-out",
+        "the file name the result will be written to ",
+        true,
+        "",
+        "output file name");
+    cmd.add(result_file);
+    cmd.parse( argc, argv );
+
+    // read surface mesh
+    std::unique_ptr<MeshLib::Mesh> surface_mesh(
+        MeshLib::IO::readMeshFromFile(in_mesh.getValue()));
+
+    std::string const prop_name("OriginalSubsurfaceNodeIDs");
+    boost::optional<MeshLib::PropertyVector<std::size_t> const&> node_id_pv(
+        surface_mesh->getProperties().getPropertyVector<std::size_t>(prop_name));
+    if (!node_id_pv)
+    {
+        ERR(
+            "Need subsurface node ids, but the property \"%s\" is not "
+            "available.",
+            prop_name.c_str());
+        return EXIT_FAILURE;
+    }
+
+    std::vector<double> integrated_values =
+        MeshLib::MeshSurfaceExtraction::getSurfaceAreaForNodes(
+            *surface_mesh, property_arg.getValue());
+    std::vector<std::pair<std::size_t, double>> direct_values;
+    direct_values.reserve(surface_mesh->getNumberOfNodes());
+
+    for (auto const* node : surface_mesh->getNodes())
+    {
+        auto const id(node->getID());
+        auto const subsurface_node_id((*node_id_pv)[id]);
+        auto const val(integrated_values[id]);
+        direct_values.push_back(std::make_pair(subsurface_node_id, val));
+    }
+
+    boost::optional<MeshLib::PropertyVector<double>&> pv(
+        surface_mesh->getProperties().createNewPropertyVector<double>(
+            "node_aggregated_gwn", MeshLib::MeshItemType::Node, 1));
```

Comment:
You get the integrated values from `property_arg`. Shouldn't the output property also be configurable? Or at least be derived from the input property?